src/urllib3/util/connection.py

```diff
@@ -60,8 +60,11 @@ def create_connection(
     except UnicodeError:
         raise LocationParseError(f"'{host}', label empty or too long") from None
 
-    for res in socket.getaddrinfo(host, port, family, socket.SOCK_STREAM):
-        af, socktype, proto, canonname, sa = res
+    try:
+        res = socket.getaddrinfo(host, port, family, socket.SOCK_STREAM)
+    except socket.gaierror as e:
+        raise e
```

Comment:
@sigmavirus24 Do you remember what you concern was in https://github.com/urllib3/urllib3/pull/1827#pullrequestreview-388532636?

Reply:
I think I'm confused as to why we're re-raising an exception with no other handling. Is it to shorten the traceback stack? What is the purpose of catching and then raising? (I'm on mobile right now so correct me if I'm missing something)

Reply:
You're right, this change itself seems useless. We could reraise as a `NameResolutionError` here instead or drop this change